crates/bdk/src/wallet/signer.rs

```diff
@@ -849,19 +903,26 @@ impl ComputeSighash for Legacy {
     }
 }
 
-fn p2wpkh_script_code(script: &Script) -> Script {
+fn p2wpkh_script_code(script: &ScriptBuf) -> ScriptBuf {
+    use bitcoin::script::PushBytes;
+    // We need to use this function instead of calling TryFrom directly
+    // on script so that Rust can figure out lifetimes correctly
+    fn script_to_pushbytes(script: &[u8]) -> &PushBytes {
+        TryFrom::try_from(script).expect("Can't fail in p2wpkh")
+    }
+    let script_bytes = script_to_pushbytes(&script[2..].as_bytes());
     ScriptBuilder::new()
         .push_opcode(opcodes::all::OP_DUP)
         .push_opcode(opcodes::all::OP_HASH160)
-        .push_slice(&script[2..])
+        .push_slice(script_bytes)
         .push_opcode(opcodes::all::OP_EQUALVERIFY)
         .push_opcode(opcodes::all::OP_CHECKSIG)
         .into_script()
 }
 
 impl ComputeSighash for Segwitv0 {
```

Comment:
I'm not exactly sure bit it looks like the implementations of this `ComputeSighash` trait contain loads of stuff that should be done by `rust-bitcoin::psbt` (complex enough to require a separate PR though). The taproot stuff is not done yet but for legacy and segwitv0 you should be able to use the new signing stuff in `psbt` (see `Psbt::sign` and `Psbt::sighash_ecdsa`). If you do play around with it and find it lacking please raise an issue.

Reply:
Fair, thanks! This PR is big enough already, but I opened https://github.com/bitcoindevkit/bdk/issues/1038 so that I can't forget :)

Reply:
Awesome, thanks.